### PR TITLE
Fix 

### PR DESCRIPTION
**Pull request by randyhoulahan**
- fix download
- fix dropbox error in view only
- pass all file varables

### CODE REVIEW

File: app/directives/print-smart/download-dialog.js [hunk 1]
```diff
@@ -54,12 +60,18 @@ define(['angular', 'lodash', 'dropbox-dropins', 'directives/checkbox'], function
...
 
     function download(event){
-      $scope.success='download'
-      if($rootScope.viewOnly){
+      
+      if($scope.viewOnly ) {
+        $scope.success='download'
+        event.preventDefault()
+        event.stopPropagation()
+        console.log(selectedLinks(true))
```
Comment (blaisefonseca) [inline app/directives/print-smart/download-dialog.js]: remove console
```suggestion
```

File: app/directives/print-smart/download-dialog.js [hunk 2]
```diff
@@ -160,9 +177,12 @@ define(['angular', 'lodash', 'dropbox-dropins', 'directives/checkbox'], function
     function closeDialogRemote(data) {
 
       var msg = data.data
-      if(msg) msg = JSON.parse(msg)
 
-      if(msg && msg.type==='closeDialogRemote')
+      if(!msg || msg !== '{"type":"closeDialogRemote"}' ) return //other events
```
Comment (sbilodeau) [inline app/directives/print-smart/download-dialog.js]: no need to test `msg !== '{"type":"closeDialogRemote"}' ` you do it later `msg.type=='closeDialogRemote'`.

File: app/directives/print-smart/download-dialog.js [hunk 3]
```diff
@@ -160,9 +176,12 @@ define(['angular', 'lodash', 'dropbox-dropins', 'directives/checkbox'], function
...
+
+      if(!_.isObject(msg)) msg = JSON.parse(msg)
```
Comment (sbilodeau) [inline app/directives/print-smart/download-dialog.js]: you should parse when it's `a string` and not when it's `not an object`

Comment (sbilodeau) [reply app/directives/print-smart/download-dialog.js]: `true` or `3` are not objects

**Comment by randyhoulahan** [comment]
@sbilodeau please review changes

**Pull request merged**